elastic/detection-rules
# [Bug] Threshold Rule Importing Failures

## Issues
https://github.com/elastic/detection-rules/issues/3547

## Summary

This PR addresses an issue where threshold rules would not import correctly as the rule prompt would ask for information that was present in the rule file instead of using what was supplied. See the original issue for the steps to reproduce the error. The fix is to remove a hardcoding loading for specifically threshold rules to load them in this way. I expect that this was done due to a historic Kibana issue that is not longer necessary. 

Note: If you import a rule and then run unit tests they will fail as we have some additional tags and fields that we enforce on our rules that should not be enforced on customers. In testing this, we also found a minor flaw in one of our unit tests (`test_event_override`) that is also fixed here. Previously the unit test would fail on the first instance of a test violation and not produce a list of all of the rules violating the test. This fix addresses that.

## Testing

1. Export a threshold rule from Kibana 
Example: [rules_export_threashold.ndjson.txt](https://github.com/elastic/detection-rules/files/14837308/rules_export_threashold.ndjson.txt)

3. Use the importer `python -m detection_rules import-rules <file.ndjson>
4. See that the importer no longer tries to prompt for threshold fields even though they exist in the ndjson
